src/libsync/propagatedownload.cpp

```diff
@@ -835,10 +835,14 @@ void PropagateDownloadFile::slotChecksumFail(const QString &errMsg, const QByteA
 
         if (cfgFile.allowChecksumValidationFail()) {
             const auto key = QString(checksumFailureDbRecordPrefix + _item->_fileId);
-            const auto numChecksumMismatchEntries = propagator()->_journal->keyValueStoreGetInt(key, 0);
-            if (numChecksumMismatchEntries < numChecksumFailuresAllowed) {
+            const QString mismatchEntryStringForFile = propagator()->_journal->keyValueStoreGet(key).toString();
+            const QStringList mismatchEntryForFileSplitted = !mismatchEntryStringForFile.isEmpty() ? mismatchEntryStringForFile.split(":") : QStringList();
+            const QByteArray mismatchChecksumForFile = mismatchEntryForFileSplitted.size() > 0 ? mismatchEntryForFileSplitted[0].toUtf8() : QByteArray("");
+            const auto numChecksumMismatchEntries = mismatchEntryForFileSplitted.size() > 1 ? mismatchEntryForFileSplitted[1].toInt() : 0;
+            if (numChecksumMismatchEntries < numChecksumFailuresAllowed || mismatchChecksumForFile != checksum) {
                 qCInfo(lcPropagateDownload) << "Checksum validation has failed" << numChecksumMismatchEntries << " times, but, allowChecksumValidationFail is set.Let's give it another try...";
-                propagator()->_journal->keyValueStoreSet(key, numChecksumMismatchEntries + 1);
+                const QString value = QString::fromUtf8(checksum) + QStringLiteral(":") + QString::number(numChecksumMismatchEntries + 1);
```

Comment:
in case `mismatchChecksumForFile != checksum` is true you will want to use 1 as the new stored value for the counter of failure otherwise you will still increase it even if checksum are always different.